cpp/src/arrow/extension/fixed_shape_tensor_test.cc

```diff
@@ -321,6 +322,70 @@ TEST_F(TestExtensionType, TestFromTensorType) {
   }
 }
 
+void CheckToTensor(const std::vector<int64_t> values, const int32_t cell_size,
```

Comment:
Also can we use something else than `int64_t` as data type? So that not all variables here have the same type...

Reply:
Added a `float32` tests.